dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/datafile/file/FileMetadataServiceTest.java

```diff
@@ -126,4 +130,40 @@ public void manageProvJson_ForPersistingProv() {
         Mockito.verify(commandEngine, Mockito.times(1)).submit(Mockito.any(PersistProvJsonCommand.class));
 
     }
+
+    @Test
+    public void findAccessibleFileMetadataSorted() {
+        //when
+        fileMetadataService.findAccessibleFileMetadataSorted(1, 1, 1);
+
+        //then
+        Mockito.verify(fileMetadataRepository, Mockito.times(1))
+               .findFileMetadataByDatasetVersionIdWithPagination(Mockito.any(Long.class),
+                                                                 Mockito.any(Integer.class),
+                                                                 Mockito.any(Integer.class));
+    }
+
+    @Test
+    public void findSearchedAccessibleFileMetadataSorted() {
+        //when
+        fileMetadataService.findSearchedAccessibleFileMetadataSorted(1, 1, 1, "");
+
+        //then
+        Mockito.verify(fileMetadataRepository, Mockito.times(1))
+               .findSearchedFileMetadataByDatasetVersionIdWithPagination(Mockito.any(Long.class),
+                                                                         Mockito.any(Integer.class),
+                                                                         Mockito.any(Integer.class),
+                                                                         Mockito.any(String.class));
+    }
+
+    @Test
+    public void findFileMetadataIds() {
+        //when
+        fileMetadataService.findFileMetadataIds(1);
+
+        //then
+        Mockito.verify(fileMetadataRepository, Mockito.times(1))
+               .findFileMetadataIdsByDatasetVersionId(Mockito.any(Long.class));
+
+    }
```

Comment:
I think it would be better to mock some results for repository and check the real results of service method invocations.